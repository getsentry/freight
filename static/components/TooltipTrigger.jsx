const React = require("react");

const $ = require("jquery");
require("bootstrap/js/tooltip");

const TooltipTrigger = React.createClass({
  propTypes: {
    title: React.PropTypes.node.isRequired,
    placement: React.PropTypes.string,
    viewport: React.PropTypes.shape({
      selector: React.PropTypes.string,
      padding: React.PropTypes.number
    })
  },

  getDefaultProps() {
    return {
      placement: "left",
      viewport: {
        selector: "body",
        padding: 5
      }
    };
  },

  componentDidMount() {
    // These can be configured via options; this is just a demo
    $(this.getDOMNode()).tooltip({
      html: true,
      placement: this.props.placement,
      title: React.renderToString(this.props.title),
      viewport: this.props.viewport
    });
  },

  componentWillUnmount() {
    var node = $(this.getDOMNode());
    node.tooltip("destroy");
    node.unbind("show.bs.tooltip", "shown.bs.tooltip", "hide.bs.tooltip", "hidden.bs.tooltip");
  },

  render() {
    return this.props.children;
  }
});

export default TooltipTrigger;
