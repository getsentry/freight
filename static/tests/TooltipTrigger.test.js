import React from 'react';
import {mount} from 'enzyme';

import TooltipTrigger from '../components/TooltipTrigger.jsx';

describe('<TooltipTrigger />', () => {
  it('should render a snapshot of <TooltipTrigger />', () => {
    const props = {
      placement: 'bottom',
      title: <div>foo</div>,
      viewport: {
        selector: 'body',
        padding: 5,
      },
    };

    const result = mount(<TooltipTrigger {...props}>{props.title}</TooltipTrigger>);
    expect(result).toMatchSnapshot();
  });
});
