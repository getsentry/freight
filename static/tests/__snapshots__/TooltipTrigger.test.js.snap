// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TooltipTrigger /> should render a snapshot of <TooltipTrigger /> 1`] = `
<TooltipTrigger
  placement="bottom"
  title={
    <div>
      foo
    </div>
  }
  viewport={
    Object {
      "padding": 5,
      "selector": "body",
    }
  }
>
  <div>
    foo
  </div>
</TooltipTrigger>
`;
